Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -994,27 +994,40 @@ const sortByIOName = (ioObject, getCustomName) => {
 };
 
 const ChannelMappingMatrix = ({ record, isShow, mapping, handleMap }) => {
-    const [expandedCols, setExpandedCols] = useState([]);
-    const [expandedRows, setExpandedRows] = useState([]);
+    const [collapsedState, setCollapsedState] = useJSONSetting(
```

Comment:
This is a good idea, except that it also persists when the user leaves this Device and is applied when the user goes to another Device's Active Map. Unlike the custom names/labels, I don't think the collapse state should be maintained per-Device, it should be reset when leaving the Device. Do you agree?

Reply:
agree but have no idea how to do it...

Reply:
OK, no problem.

Reply:
there is easy way to do it when leaving the active map tab.
```
 <Route
      exact path={`${props.basePath}/${props.id}/show/active_map`}
      onLeave={ removeItem }
    >
```

Reply:
what do you think ?

Reply:
sounds reasonable, I think